src/main/java/ay2021s1_cs2103_w16_3/finesse/model/budget/ObservableCurrency.java

```diff
@@ -0,0 +1,73 @@
+package ay2021s1_cs2103_w16_3.finesse.model.budget;
+
+import static ay2021s1_cs2103_w16_3.finesse.commons.util.AppUtil.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+import java.math.BigDecimal;
+
+import javafx.beans.property.ObjectProperty;
+import javafx.beans.property.SimpleObjectProperty;
+
+/**
+ * An abstract class representing currency values.
+ * Guarantees: immutable; is valid as checked by the Amount class
+ */
+public abstract class ObservableCurrency {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Amounts should only contain numbers, with an optional 2 decimal places or '$' prefix";
+    public static final String VALIDATION_REGEX = "^\\$?\\d+(\\.\\d{2})?$";
+
+    private BigDecimal value;
```

Comment:
Instead of duplicating code from `Amount`, perhaps you could write this to encapsulate an `Amount` instead?

This should solve the validation and abstraction issues faced in the other classes.

Reply:
Also, avoid dealing directly with `BigDecimal` as it will break if we decide to change the underlying data type for `Amount`. If there's anything you need from the underlying value, just add the functionality to `Amount`.